proposals/accepted/000-ast-parser-libs.rst

```diff
@@ -41,33 +41,69 @@ This section highlights various past and ongoing efforts accordingly.
+see the *hundreds* of modules included inside it, many of which have nothing to do with the Haskell surface language.
+All this `"bycatch" <https://en.wikipedia.org/wiki/Bycatch>`_ in the extraction process results in a library that daunting to use, and which has a hard time presenting a stable interface.
+
+The lesson from |haskell-src-exts| are clear:
+
+- it succeeded in being modular and self-contained but failed trying to keep up with GHC;
+
+- given the size of the community, competing head-on with GHC or trying to keep up with it is very difficult:
+
+  - being used by GHC, so keeping up happens automatically, is the clearest way to avoid this problem.
 
-The developers of ``ghc-lib-parser`` would not dispute the above criticisms, for this state of affairs was never intended to be a permanent solution, but rather just a stop gap.
-Whereas ``ghc-lib-parser`` succeeds in keeping up with GHC because it *is* GHC, it fails in being self-contained because modularity cannot be `"fixed in post" [production] <https://tvtropes.org/pmwiki/pmwiki.php/Main/FixItInPost>`_.
+Whereas |ghc-lib-parser| succeeds in keeping up with GHC because it *is* GHC, it fails in being self-contained because modularity cannot be `"fixed in post" [production] <https://tvtropes.org/pmwiki/pmwiki.php/Main/FixItInPost>`_.
 Code that is intended to be separate from any one consumer must be developed with those boundaries enforced during development.
 
+.. [#ghcinception] The extraction process was enabled by insights gained from the `"GHCinception" <https://mgsloan.com/posts/ghcinception/>`_ or "GHC in GHCi" initative.
+
+Downstream projects
+-------------------
+
+In addition to going over the major AST/parser libraries for Haskell, it is also useful to talk about the most notable projects that use them.
+Ultimately, it is those projects we want to help out.
+
+HLint_, the Haskell linter project developed continuously since 2008 is the most notable one, not just because its longstanding and wide use, but also because many of the developers that worked on the previous two libraries also worked on it --- use by HLint served as proof the libraries were fit-for-purpose.
+
+In June 2019,
+following the release of |ghc-lib-parser| and deprecation of |haskell-src_exts| earlier that year as described above,
+`HLint began the transitition <http://neilmitchell.blogspot.com/2019/06/hlints-path-to-ghc-parser.html>`_ to |ghc-lib-parser|.
+In May 2020, the release of HLint-3.0 which "uses the GHC parser" `was announced <http://neilmitchell.blogspot.com/2020/05/hlint-30.html>`_.
+
+Today, most users of |haskell-src-exts| have largely migrated from |haskell-src-exts| to |ghc-lib-parser| [#exampleghclibparserusers]_.
+But just because all these projects are using |ghc-lib-parser| doesn't mean everything is well.
+**Insert quote about maintainence overhead.**
```

Comment:
Thinking of what you put in the email to me about how much time the "ghc-lib-parser/ghc-lib/ghc-lib-parser-ex/hlint stack" was costing you, @shayne-fletcher.

Reply:
that's kind of you john. feel free to adapt "(ghc-lib-parser related maintenance) generally costs an hour or two a week and often more" in any way you see fit!

Reply:
It's not just kind, it is **crucial evidence** the modularity question isn't just an abstract one of code quality / library friendliness, but also a concrete one of reducing wasteful costs that you are stuck paying today!